### DISCUSSION TITLE
Show single asset shares on the web instead of instantly downloading it

### DISCUSSION THREAD

**Discussion by JordyEGNL** [alangrainger/immich-public-proxy#10] [Feature Requests]
When you share a single image with Immich you can view it first on the web and then choose to download it (or only view it)

With Immich public proxy the image will be downloaded instantly and the visitor cannot see the asset in advance

**Comment by alangrainger** [comment]
Yes that's the intended behaviour, to allow image embedding (for example adding to a blog or automatically unfurling when sharing via a messaging app). Per [the docs](https://github.com/alangrainger/immich-public-proxy/blob/main/README.md#features):

> If sharing a single image, the link will directly open the image file so that you can embed it anywhere you would a normal image.

I can make that configurable behaviour.

**Reply by JordyEGNL** [reply]
Upon further investigation it seems that Immich public proxy downloads a .bin file thus making it not viewable on any device.

Sharing albums works great though :)

![Screenshot_20241103-105826_Firefox.png](https://github.com/user-attachments/assets/ab10274f-44c4-44b1-8a85-bac43551ffdc)

![Screenshot_20241103-105814_Firefox~2.png](https://github.com/user-attachments/assets/41a48968-3195-46d2-956f-02187e38d507)

**Reply by JordyEGNL** [reply]
Direct linking that url into a HTML<src> element works as you described

It would be great if this is a toggleable feature

**Reply by alangrainger** [reply]
Thanks - that's a bug, I can see the `content-type` is coming through as undefined for some reason.

Issue added here: #11

**Reply by alangrainger** [reply]
Thanks for spotting the `content-type` issue @JordyEGNL - this is now fixed in v1.3.5.

I will look to add the individual image "direct or gallery" configuration next week.

**Reply by JordyEGNL** [reply]
Thanks for the quick response 🤩

**Comment by alangrainger** [comment]
Configuration option for this is added in v1.3.7.

See [Additional configuration](https://github.com/alangrainger/immich-public-proxy?tab=readme-ov-file#additional-configuration) and adjust the settings in `config.json`. Specifically, change `singleImageGallery` to `true`.

**Reply by JordyEGNL** [reply]
Thanks I got it working! This is my config.json
```json
{
  "ipp": {
    "responseHeaders": {
      "Cache-Control": "public, max-age=86400"
    },
    "singleImageGallery": true
  },
  "lightGallery": {
    "controls": true,
    "download": false,
    "mobileSettings": {
      "controls": true,
      "showCloseIcon": true,
      "download": false
    }
  }
}
```
